state/endpoint_bindings.go

```diff
@@ -84,13 +84,13 @@ func (b bindingsMap) GetBSON() (interface{}, error) {
 func (b *Bindings) Merge(mergeWith map[string]string, meta *charm.Meta) (bool, error) {
 	// Verify the bindings to be merged, and ensure we're merging with
 	// space ids.
-	merge, err := NewBindings(b.st, mergeWith)
+	merge, err := NewBindings(b.allSpaces, mergeWith)
```

Comment:
I think we might have saved a lot of churn by removing `newBindingsFromNames` (it was for backwards compatibility), and just assuming that supplied binding IDs were valid.